modules/commons/src/main/java/org/apache/synapse/commons/vfs/VFSUtils.java

```diff
@@ -373,6 +373,17 @@ public static boolean isFailRecord(FileSystemManager fsManager, FileObject fo) {
         return false;
     }
 
+    public static boolean canBeRead(FileObject fo, Long waitTimeBeforeRead) {
+        if(waitTimeBeforeRead != null) {
+            try {
+                return fo.getContent().getLastModifiedTime() < (new Date().getTime() - waitTimeBeforeRead);
```

Comment:
Is it possible to get time from System?
does fo.getContent().getLastModifiedTime() return in ms?

Reply:
I have changed to get the time from System.